com.playeveryware.eos/Runtime/Core/Config/ProductConfig.cs

```diff
@@ -114,6 +141,86 @@ protected override bool NeedsMigration()
         
         protected ProductConfig() : base("eos_product_config.json") { }
 
+        protected override void OnReadCompleted()
+        {
+            // This tracks whether there is a single deployment defined.
+            _deploymentDefined = Environments.IsDeploymentDefined;
+        }
+
+        // This compile conditional is here because the OnWriteCompleted method
+        // is only defined in the editor - so it can only be overriden if in the
+        // editor.
+#if UNITY_EDITOR
+
+        protected override void BeforeWrite()
+        {
+            // If there is one deployment, and one sandbox, then make sure they
+            // are linked to each other.
+            // But only do this if they have been newly added
+            if (!_deploymentDefined && Environments.Deployments.Count ==1 && Environments.Sandboxes.Count == 1)
+            {
+                Environments.Deployments[0].Value.SandboxId = Environments.Sandboxes[0].Value;
+            }
+        }
+
+        protected override void OnWriteCompleted()
+        {
+            // If when the config was last read there was a deployment defined,
+            // or there is not now one defined - then there is no need to try
+            // and set the deployment values for each platform config.
+            if (_deploymentDefined || !Environments.IsDeploymentDefined)
+            {
+                return;
+            }
+
+            // Select the first defined deployment as the deployment to set
+            // platform configs to use.
+            Named<Deployment> deploymentToSetPlatformsTo = Environments.Deployments[0];
```

Comment:
This should be fixed so that it is not assumed that the correctly defined deployment is the first one. Perhaps have a "GetFirstDefinedDeployment" function?

Reply:
Implementation was updated to use a newly defined function called `TryGetFirstNamedDeployment`.

Accordingly, I am going to mark this conversation as resolved.